American Whitewater/ViewControllers/OnBoard/OnboardLocationViewController.swift

```diff
@@ -90,79 +88,72 @@ class OnboardLocationViewController: UIViewController, CLLocationManagerDelegate
+    private func updateDefaults(coordinate: CLLocationCoordinate2D, regionCodes: [String]? = nil) {
+        var filters = Filters.defaultByDistanceFilters
+        
+        // Set the region code if available, but otherwise avoid clobbering anything that was already set
+        filters.regionsFilter = regionCodes ?? DefaultsManager.shared.filters.regionsFilter
+        
+        DefaultsManager.shared.filters = filters
+        DefaultsManager.shared.coordinate = coordinate
+        DefaultsManager.shared.onboardingCompleted = true
+        
+        // FIXME: why is this the onboarding controller's job?
+        DefaultsManager.shared.appVersion = Double( (Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String) ?? "" ) ?? 0.0
+    }
+    
+    private func regionCodes(placemarks: [CLPlacemark]?) -> [String]? {
+        guard
+            let placemark = placemarks?.first,
+            let adminArea = placemark.administrativeArea,
+            let region = Region.regionByCode(code: "st\(adminArea)")
+        else {
+            return nil
+        }
+        
+        return [region.code]
+    }
+}
+
+extension OnboardLocationViewController: CLLocationManagerDelegate {
     func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
         guard let location = locations.last else {
             return
         }
-
-        print("Got the latest location: \(location.debugDescription)")
-        // stop updating
+        
         locationManager.stopUpdatingLocation()
         
-        // store the location for future use
-        DefaultsManager.shared.coordinate = location.coordinate
+        self.updateDefaults(
+            coordinate: location.coordinate,
+            regionCodes: nil
+        )
         
-        DefaultsManager.shared.filters = Filters.defaultByDistanceFilters
-
-        DefaultsManager.shared.onboardingCompleted = true
-        DefaultsManager.shared.appVersion = Double( (Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String) ?? "" ) ?? 0.0
-        
-        // reverse geocode the users location so we can get the admin area so we can request this location first
-        let decoder = CLGeocoder()
-        decoder.reverseGeocodeLocation(location) { (placemarks, error) in
-            if
-                error == nil,
-                let placemark = placemarks?.first,
-                let adminArea = placemark.administrativeArea,
-                let region = Region.regionByCode(code: "st\(adminArea)")
-            {
-                // FIXME: this updates the regions filter after dismissing the view controller, notifying the parent. Is this a problem?
-                DefaultsManager.shared.filters.regionsFilter = [region.code]
+        // Reverse geocode to set the regionsFilter
+        CLGeocoder().reverseGeocodeLocation(location) { [weak self] (placemarks, error) in
```

Comment:
Shouldn't this reverse geocode happen before updating defaults? Seems like it would be simpler to update everything at once.